docs/naps/4-async-slicing.md

```diff
@@ -559,6 +559,59 @@ event per layer that only contains that layer's slice response.
 - Likely need an `asyncio` event loop distinct from Qt's main event loop, which could be confusing and cause issues.
 
 
+### Slice from vispy layer instead of viewer model
+
+Instead of making the `ViewerModel` the driver of slicing, we could instead drive it from the vispy layers.
+A rough implementation of this approach could look like the following.
+
+```python
+
+class QtViewer:
+    ...
+    def __init__(self):
+        ...
+        self.viewer.dims.current_step.connect(self.slice_layers)
+
+    def slice_layers(self):
+        for layer in self.viewer.layers:
+            self.layer_to_visual[layer]._slice(self.viewer.dims)
+
+class Image:
+    ...
+    def _make_slice_request(self, dims: Dims) -> ImageSliceRequest:
+        ...
+    def _get_slice(self, request: ImageSliceRequest) -> ImageSliceResponse:
+        ...
+
+class VispyImageLayer:
+    ...
+    def _slice(self, dims: Dims) -> None:
+        request = self.layer._make_slice_request(dims)
+        task = self.slice_executor.submit(self.layer._get_slice, request)
+        task.add_done_callback(self._on_slice_done)
+
+    def _on_slice_done(self, task: Future[ImageSliceResponse]) -> None:
+        self._set_slice(task.result())
+
+    @ensure_main_thread
```

Comment:
One small remark. This introduces Qt dependency outside `_qt` directory. I understand that it could be a generalization. But if we decide to introduce Qt to vispy module, then we could just make `VispyImageLayer` A Qt Object and direct use signals that have better performance and may simplify the problem of synchronized mode.

Reply:
Good point. I added a new paragraph in this section to capture that idea.